staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -293,6 +293,17 @@ func (t *tracker) Get(gvr schema.GroupVersionResource, ns, name string) (runtime
 		return nil, fmt.Errorf("more than one object matched gvr %s, ns: %q name: %q", gvr, ns, name)
 	}
 
+	acc, err := meta.Accessor(matchingObjs[0])
+	if err != nil {
+		return nil, fmt.Errorf("unable to get meta accessor: %v", err)
+	}
+	if acc.GetNamespace() != ns {
```

Comment:
Can you add a comment saying why we need to check the namespace and name again after we have called `filterByNamespaceAndName`?

Reply:
Done